### ISSUE TITLE
Flaky test: TestBootstrapTokenRotationMachinePool

### ISSUE THREAD

**Issue by mkumatag** [kubernetes-sigs/cluster-api#5724]
**What steps did you take and what happened:**
While testing the https://github.com/kubernetes-sigs/cluster-api/pull/5662, we have seen this test flaking, report can be found [here](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/5662/pull-cluster-api-test-main/1463398081641844736)

```shell
=== RUN   TestBootstrapTokenRotationMachinePool
I1124 06:46:21.977411   12139 kubeadmconfig_controller.go:796]  "msg"="Altering JoinConfiguration.Discovery.BootstrapToken"  "APIServerEndpoint"="100.105.150.1:6443"
I1124 06:46:21.980192   12139 kubeadmconfig_controller.go:812]  "msg"="Altering JoinConfiguration.Discovery.BootstrapToken"  
I1124 06:46:21.980621   12139 kubeadmconfig_controller.go:516]  "msg"="Creating BootstrapData for the worker node" "kind"="MachinePool" "name"="worker-machinepool" "version"="999" 
I1124 06:46:22.985599   12139 kubeadmconfig_controller.go:298]  "msg"="Refreshing token until the infrastructure has a chance to consume it"  
I1124 06:46:23.990471   12139 kubeadmconfig_controller.go:328]  "msg"="Altering JoinConfiguration.Discovery.BootstrapToken"  "Token"="ary5z5.cd4bnlevzi15bte1"
I1124 06:46:23.991091   12139 kubeadmconfig_controller.go:516]  "msg"="Creating BootstrapData for the worker node" "kind"="MachinePool" "name"="worker-machinepool" "version"="1000" 
I1124 06:46:24.000209   12139 kubeadmconfig_controller.go:904]  "msg"="bootstrap data secret for KubeadmConfig already exists, updating"  "KubeadmConfig"="workerpool-join-cfg" "secret"="workerpool-join-cfg"
    kubeadmconfig_controller_test.go:1000: 
        Expected
            <string>: 2021-11-24T07:01:23Z
        to equal
            <string>: 2021-11-24T07:01:24Z
--- FAIL: TestBootstrapTokenRotationMachinePool (3.09s)
```

Seen sometime in the past as well: https://storage.googleapis.com/k8s-triage/index.html?test=TestBootstrapTokenRotationMachinePool

**What did you expect to happen:**

This is unit test and should pass all the time

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

/kind bug
/kind flake
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

**Comment by fabriziopandini** [comment]
/milestone v1.1

@mkumatag do you want to give it a go at this?

**Comment by mkumatag** [comment]
> /milestone v1.1
> 
> @mkumatag do you want to give it a go at this?

may need some time to understand the testcase, can give it a try :)

**Comment by mkumatag** [comment]
This is happening because the validation is comparing the expiration TTL with the TTL calculated based on the time.Now() + TTL, this may produce inconsistent results because there is always a lag between the token rotation in the reconcile method and the test case(this happens when the lag is more than a second).

So the best way to fix this issue is to check that token expiry is between the just before rotate TTL and time.Now() TTL and fixed is pushed via #5733 

@fabriziopandini
